golos.vesting/golos.vesting.cpp

```diff
@@ -541,5 +545,32 @@ void vesting::timeout_delay_trx() {
     trx.send(_self.value, _self);
 }
 
+void vesting::paydelegator(name voter, asset reward, name delegator, 
+        uint64_t interest_rate, uint8_t payout_strategy) {
+    if (payout_strategy == config::to_delegated_vesting) {
+        tables::account_table acc_table_dlg(_self, delegator.value);
+        auto balance_dlg = acc_table_dlg.find(reward.symbol.code().raw());
+        acc_table_dlg.modify(balance_dlg, voter, [&](auto& item) {
+            item.delegate_vesting += reward;
+        });
+        tables::account_table acc_table_rcv(_self, voter.value);
+        auto balance_rcv = acc_table_rcv.find(reward.symbol.code().raw());
+        acc_table_rcv.modify(balance_rcv, delegator, [&](auto& item) {
+            item.received_vesting += reward;
+        });
+        tables::delegate_table table(_self, reward.symbol.code().raw());
+        auto index_table = table.get_index<"unique"_n>();
+        auto delegate_record = index_table.find(structures::delegate_record::unique_key(delegator, voter));
+        index_table.modify(delegate_record, name(), [&](auto& item) {
+            item.deductions += reward;
+        });
+    } else if (payout_strategy == config::to_delegator) {
+        tables::account_table acc_table(_self, delegator.value);
+        auto balance = acc_table.find(reward.symbol.code().raw());
+        acc_table.modify(balance, voter, [&](auto& item) {
+            item.delegation_rewards += reward;
```

Comment:
Why do we need separate field?
In Golos it's for statistics, and don't used in consensus. I think we don't need this field in current task, and in place of it should be some event later.

The 1st step for both payment strategies is to increase delegator's vesting balance. If strategy is `to_delegated_vesting` and there is delegation object exists (`delegate_record`), then update delegation object and related `delegate_vesting` + `received_vesting`